core/includes/image.inc

```diff
@@ -577,30 +577,39 @@ function image_is_svg($uri) {
  *   which is the accessible mechanism for alternative text within SVGs.
  *
  * @return string
- *   The SVG image contents with the attributes added.
+ *   The SVG image contents with the attributes added. An empty string in case
+ *   of errors.
  *
  * @since 1.28.0 Function added.
  */
 function image_add_svg_attributes($svg_content, array $attributes) {
   $doc = new DOMDocument();
+  libxml_use_internal_errors(TRUE);
   $doc->loadXML($svg_content);
   $svg_tag = $doc->getElementsByTagName('svg')->item(0);
+  $last_error = libxml_get_last_error();
+  libxml_use_internal_errors(FALSE);
+  if (empty($svg_tag)) {
+    $message = $last_error ? trim($last_error->message) : '';
+    watchdog('image', 'Failed to parse SVG content. %message', array(
+      '%message' => $message,
+    ), WATCHDOG_ERROR);
+    return '';
+  }
 
   // Convert the alt attribute to a <title> element.
   if (isset($attributes['alt'])) {
-      try {
-        if (strlen($attributes['alt'])) {
-          $title = $doc->createElement('title');
-          $title->textContent = $attributes['alt'];
-          // Since DOMDocument::prepend() is not available in PHP versions prior
-          // to v8, we are using DOMNode::insertBefore().
-          $svg_tag->insertBefore($title, $svg_tag->firstChild);
-        }
-        // Remove any given <title> element if alt is an empty string.
-        elseif ($svg_tag->firstChild && $svg_tag->firstChild->nodeName === 'title') {
-          $svg_tag->removeChild($svg_tag->firstChild);
-        }
-      } catch (DOMException $e) {}
+    if (strlen($attributes['alt'])) {
+      $title = $doc->createElement('title');
+      $title->textContent = $attributes['alt'];
+      // Since DOMDocument::prepend() is not available in PHP versions prior
+      // to v8, we are using DOMNode::insertBefore().
+      $svg_tag->insertBefore($title, $svg_tag->firstChild);
+    }
+    // Remove any given <title> element if alt is an empty string.
+    elseif ($svg_tag->firstChild && $svg_tag->firstChild->nodeName === 'title') {
+      $svg_tag->removeChild($svg_tag->firstChild);
+    }
```

Comment:
I realize that this is not part of the scope here, but is this whole code here making an assumption that the given svg does not have an existing `<title>`? 🤔

Reply:
Probably yes - should we consider that? We could...

Reply:
Currently the only usage of that function is in icon.inc - I'd assume, that icons never ship with a title tag out-of-the-box.
However, this function *could* in theory be used on other SVG files, too. In that case a check for existing title makes sense. But what is the expected outcome then? Overriding or leaving the original untouched?

Reply:
@klonos  As this is out of scope here, but a valid concern, which deserves a little more discussion, I've created a follow-up on https://github.com/backdrop/backdrop-issues/issues/6731
Let's continue to discuss desired behavior there.